iPetCare/Application/Services/NoteService.cs

```diff
@@ -0,0 +1,190 @@
+                return new ServiceResponse<GetNoteDtoResponse>(HttpStatusCode.Unauthorized);
+
+            var pet = await Context.Pets.FindAsync(petId);
+
+            if (pet == null)
+                return new ServiceResponse<GetNoteDtoResponse>(HttpStatusCode.BadRequest, "Nie znaleziono zwierzaka");
+
+            if (!CanEditNote(pet))
+                return new ServiceResponse<GetNoteDtoResponse>(HttpStatusCode.Forbidden);
+
+            var note = await Context.Notes.FindAsync(noteId);
+            if (note == null)
+                return new ServiceResponse<GetNoteDtoResponse>(HttpStatusCode.NotFound);
+
+            var dto = Mapper.Map<GetNoteDtoResponse>(note);
+
+            return new ServiceResponse<GetNoteDtoResponse>(HttpStatusCode.OK, dto);
+        }
+
+        public async Task<ServiceResponse<GetAllNotesDtoResponse>> GetPetNotesAsync(Guid petId)
+        {
+            if (CurrentlyLoggedUser == null)
+                return new ServiceResponse<GetAllNotesDtoResponse>(HttpStatusCode.Unauthorized);
+
+            var pet = await Context.Pets.FindAsync(petId);
+
+            if (pet == null)
+                return new ServiceResponse<GetAllNotesDtoResponse>(HttpStatusCode.NotFound);
+
+            if (!CanEditNote(pet))
+                return new ServiceResponse<GetAllNotesDtoResponse>(HttpStatusCode.Forbidden);
+
+            var notes = await Context.Notes.Where(ex => ex.PetId == petId).ToListAsync();
+
+            var dto = new GetAllNotesDtoResponse()
+            {
+                Notes = Mapper.Map<List<NoteForGetAllNotesDtoResponse>>(notes)
+            };
+
+            return new ServiceResponse<GetAllNotesDtoResponse>(HttpStatusCode.OK, dto);
+        }
+
+        public async Task<ServiceResponse<UpdateNoteDtoResponse>> UpdateNoteAsync(Guid petId, Guid noteId, UpdateNoteDtoRequest dto)
+        {
+            if (CurrentlyLoggedUser == null)
+                return new ServiceResponse<UpdateNoteDtoResponse>(HttpStatusCode.Unauthorized);
+
+            var pet = await Context.Pets.FindAsync(petId);
+            if (pet == null)
+                return new ServiceResponse<UpdateNoteDtoResponse>(HttpStatusCode.BadRequest, "Nie znaleziono zwierzaka");
+
+            if (noteId == Guid.Empty)
+                return new ServiceResponse<UpdateNoteDtoResponse>(HttpStatusCode.BadRequest, "Nieprawidłowy indeks");
+
+            if (!CanEditNote(pet))
+                return new ServiceResponse<UpdateNoteDtoResponse>(HttpStatusCode.Forbidden);
+
+            var note = await Context.Notes.FindAsync(noteId);
+
+            if (note == null)
+                return new ServiceResponse<UpdateNoteDtoResponse>(HttpStatusCode.NotFound);
+
+            var user = await Context.Users.FindAsync(dto.UserId);
+
+            if (user == null)
+                return new ServiceResponse<UpdateNoteDtoResponse>(HttpStatusCode.BadRequest, "Nie znaleziono typu użytkownika");
+           
+            note.Payload = dto.Payload;
```

Comment:
Odwieczny problem edycji, jeśli ktoś zmienił wszystko na dokładnie taki sam stan jak był, to czy to jest błąd czy nie? Bo EF tego nie wychwyci i savechanges zwróci 0, czyli jakby błąd.
Może dobrze byłoby sprawdzić, czy wszystko jest dokładnie takie samo jak było, jak tak to zwrócić okejkę :D Tzn ogólnie to pierwszą warstwą ochronną powinien być front, ale wiadomo jak to jest z zabezpieczeniami po stronie klienta :D 

Oczywiście nie jest to stricte błąd, nic się nie wywali, po prostu zwrócony zostanie badrequest mimo, że wszystko się powiodło :)